src/main/java/org/opennars/inference/TruthFunctions.java

```diff
@@ -23,6 +23,33 @@
  * All truth-value (and desire-value) functions used in inference rules 
  */
 public final class TruthFunctions extends UtilityFunctions {
+    public enum EnumType {
+        DESIREDED,
+        DESIREIND,
+        DESIREWEAK,
+        DESIRESTRONG,
+        COMPARISON,
+        ANALOGY,
+        ANONYMOUSANALOGY,
+    }
+
+    public static TruthValue dispatchBinary(final EnumType type, final TruthValue a, final TruthValue b) {
+        switch(type) {
+            case DESIREDED: return desireDed(a, b);
+            case DESIREIND: return desireInd(a, b);
+            case DESIREWEAK: return desireWeak(a, b);
+            case DESIRESTRONG: return desireStrong(a, b);
+            case COMPARISON: return comparison(a, b);
+            case ANALOGY: return analogy(a, b);
+            case ANONYMOUSANALOGY: return anonymousAnalogy(a, b);
+            default: throw new RuntimeException("Encountered unimplemented case!"); // internal error
+        }
+    }
+
+    public static TruthValue dispatchBinary2(final boolean v, final EnumType typeTrue, final EnumType typeFalse, final TruthValue a, final TruthValue b) {
```

Comment:
dont name this `dispatchBinary2` just name it `dispatchBinary` and use overloading.

Reply:
Overloading is confusing here, how to name this?

Reply:
Same way you have it named now just remove the 2 from the other one... Am I misunderstanding the question?